tools/cldr-code/src/main/java/org/unicode/cldr/util/LocalesTxtReader.java

```diff
@@ -0,0 +1,205 @@
+                    }
+                }
+
+                Level status = Level.get(stuff.get(2));
+                if (status == Level.UNDETERMINED) {
+                    System.out.println("Warning: Level unknown on: " + line);
+                }
+                Map<String, Level> locale_status = platform_locale_level.get(organization);
+                if (locale_status == null) {
+                    platform_locale_level.put(organization, locale_status = new TreeMap<>());
+                }
+                locale_status.put(locale, status);
+                if (!locale.equals(StandardCodes.ALL_LOCALES)) {
+                    String scriptLoc = parser.getLanguageScript();
+                    if (locale_status.get(scriptLoc) == null)
+                        locale_status.put(scriptLoc, status);
+                    String lang = parser.getLanguage();
+                    if (locale_status.get(lang) == null)
+                        locale_status.put(lang, status);
+                }
+
+                if (weight != null) {
+                    organization_locale_weight
+                        .computeIfAbsent(organization, ignored -> new TreeMap<String, Integer>())
+                        .put(locale, weight);
+                }
+                if (pathMatch != null) {
+                    organization_locale_match
+                        .computeIfAbsent(organization, ignored -> new TreeMap<String, Set<String>>())
+                        .put(locale, ImmutableSet.copyOf(pathMatch.split(",")));
+                }
+            }
+        } catch (IOException e) {
+            throw new ICUUncheckedIOException("Internal Error", e);
+        }
+
+        // now reset the parent to be the max of the children
+        for (Organization platform : platform_locale_level.keySet()) {
+            Map<String, Level> locale_level = platform_locale_level.get(platform);
+            for (String locale : locale_level.keySet()) {
+                parser.set(locale);
+                Level childLevel = locale_level.get(locale);
+
+                String language = parser.getLanguage();
+                if (!language.equals(locale)) {
+                    Level languageLevel = locale_level.get(language);
+                    if (languageLevel == null || languageLevel.compareTo(childLevel) < 0) {
+                        locale_level.put(language, childLevel);
+                    }
+                }
+                String oldLanguage = language;
+                language = parser.getLanguageScript();
+                if (!language.equals(oldLanguage)) {
+                    Level languageLevel = locale_level.get(language);
+                    if (languageLevel == null || languageLevel.compareTo(childLevel) < 0) {
+                        locale_level.put(language, childLevel);
+                    }
+                }
+            }
+        }
+        // backwards compat hack
```

Comment:
Can you explain what is going on here, and why we need this?

Reply:
I will see if I can figure it out, but this function was moved from StandardCodes.java

Reply:
Ok, if it is just copied that's fine for now.

Reply:
filed subtask CLDR-15625 for straightening this out